src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -154,6 +155,8 @@ static Popup()
         /// </summary>
         public event EventHandler? Opened;
 
+        internal event EventHandler<CancelEventArgs>? Closing;
```

Comment:
Any better ideas how we can cancel popup closing? To solve https://github.com/AvaloniaUI/Avalonia/issues/6057
I don't want to add public API event, since neither WPF nor UWP has it.

Reply:
An internal event works for me for now.